core/go/internal/publictxmgr/in_flight_transaction_stage_controller.go

```diff
@@ -135,10 +138,21 @@ func NewInFlightTransactionStageController(
 
 	ift.MarkTime("wait_in_inflight_queue")
 	imtxs := NewInMemoryTxStateManager(enth.ctx, ptx)
-	ift.stateManager = NewInFlightTransactionStateManager(enth.thMetrics, enth.balanceManager, enth.bIndexer, ift, imtxs, oc.retry, oc, oc.submissionWriter, ift.testOnlyNoEventMode)
+	ift.stateManager = NewInFlightTransactionStateManager(enth.thMetrics, enth.balanceManager, ift, imtxs, oc, oc.submissionWriter, ift.testOnlyNoEventMode)
 	return ift
 }
 
+func (it *inFlightTransactionStageController) UpdateTransaction(ctx context.Context, newPtx *DBPublicTxn) {
+	it.updateMux.Lock()
+	defer it.updateMux.Unlock()
+
+	// Updates queued to be handled in the main orchestrator loop function so that all code that consumes and modifies the transaction
+	// values is in the same goroutine. There is some risk deferring the response to a synchronous API call to an asynchronous loop
+	// function; however, that function starts any action with higher latency in separate go routines meaning that we can expect
+	// ProduceLatestInFlightStageContext to run with an acceptable frequency for the response.
+	it.updates = append(it.updates, newPtx)
```

Comment:
Ok - so the actual DB update action is deferred, and we don't wait here for the DB change to be applied before we return.

I think it's the latter that will be surprising to people.

So I'd like your thought on why we don't:
- Wrap the `*DBPublicTxn` in a struct that has a `done chan error` on it before we put it on updates
- Drop the lock
- Wait for `<-update.done` and return an error (with context based timeout)

This would allow us to:
- Ensure that the caller can do a query immediately after the call, and see whatever the change is applied
- Return an error if the update is not valid/relevant at the point it's picked up by the main routine